c_wrapper/lua-mongoc-collection.c

```diff
@@ -650,53 +650,54 @@ lua_mongo_collection_create_index(lua_State *L)
     int keys_index = 3;
     int options_index = 4;
 
-    if (!(lua_istable(L, keys_index))) {
+    if(!(lua_istable(L, keys_index))){
         luaL_error(L, "keys parameter must be a table");
-    }
-
-    if (!(lua_isnil(L, keys_index))){
+    }else{
```

Comment:
This line will also cause memory to be allocated, since you don't call `bson_free` on it. [docs](http://mongoc.org/libmongoc/1.1.2/mongoc_collection_keys_to_index_string.html).

We're going to have to track this with a boolean.
```
int
lua_mongo_collection_create_index(lua_State *L)
{
  bool used_default_index_name;

  ...

  used_default_index_name = !lua_isstring(L, -9);
  if(!used_default_index_name){

  ...

DONE:
  if(used_default_index_name){
    bson_free(opts.name);
  }
}
```